mantle/platform/api/azure/instance.go

```diff
@@ -156,7 +156,12 @@ func (a *API) CreateInstance(name, userdata, sshkey, resourceGroup, storageAccou
 
 	vmParams := a.getVMParameters(name, userdata, sshkey, fmt.Sprintf("https://%s.blob.core.windows.net/", storageAccount), ip, nic)
 
-	_, err = a.compClient.CreateOrUpdate(resourceGroup, name, vmParams, nil)
+	// this line is prone to hangs, so lets wrap it and run it in a timeout aware
+	// manner
+	err = util.RetryWithTimeoutAsync(5*time.Minute, 1, 10*time.Second, func() error {
```

Comment:
If we're only trying once, do we need a retry function at all?

Reply:
I think the idea was to make the `RetryWithTimeoutAsync` function generic enough to handle retries like the other helpers in `retry.go`.